src/controllers/helpers/pagination.rs

```diff
@@ -24,6 +28,10 @@ impl Page {
                 )));
             }
 
+            if numeric_page > MAX_PAGE_BEFORE_SUSPECTED_BOT {
+                req.log_metadata("bot", "suspected");
```

Comment:
I have to say I'm not a big fan of having to pass in the `RequestExt` here just so that we can use the metadata logging functionality. We might want to have a look into alternatives at some point that allow for proper structured logging (e.g. the `tracing` crate).